### ISSUE TITLE
My thoughts after using this mod

### ISSUE THREAD

**Issue by TaroooLvr** [BudgetAirpods/InsanityRemastered#8]
After playing the mod I instantly loved how it worked, it made the game fun but at the same time, I encountered some problems that I didn't like while using the mod. So my list is:
1. The fake items are too repetitive and they keep appearing in my games, make them appear less or have a config for them. Much better if we could control how many of the 3 hallucinations we get.
2. The sounds are too loud. The sounds are good, but it doesn't really fit on the modpack I was using and because of that, we just kept hearing a ton of random sounds. We deleted the sounds that were with the mod because we think it is unnecessary. Though does that mean we won't get some audio hallucinations? because we disabled it right after deleting the sounds.
3. Sanity gain and sanity loss is weird. I easily get max insanity after going to the facility for a short duration. I even have changed the config to have high sanity gain and less sanity loss (in which what happened is I get that insanity screen when I go away from my friends even for like 5 meters or something and if I go back to them again, I lose that screen. It's quite unnatural)
4. even without monsters, exploring deep in the facility gives me the "insanity" screen like if you see a monster or a dead player. I don't like this because we play with a person on the terminal acting as our guide, so we trust who has that role. So instead of the mod making the game scary, it just makes it distracting because of that screen.
5. have like a bar or indicator for how much sanity you've gained or lost
We love playing with this mod because of how much opportunity it has. Heck, I love this mod as much as the Diversity mod because it only changes and enhances the vanilla aspects of the game. also, right now we only use the mod to experience the model hallucinations and the panic attacks, but we really wanna use all of the mod.

**Comment by BudgetAirpods** [comment]
1. I do plan on making them less common as I add more hallucinations. They serve as a "buildup" for the other hallucinations. They are toggleable in the config file if they are really bothering you.

2. I'm going to lower the volume of a lot of sounds and even reduce the length of some of them soon. SoundResources_SFX contains the custom sounds of the mod, while SoundResources_LCGame contains sounds from the vanilla game. If you really want, you can remove the SFX one and have the vanilla sounds only. (Keep in mind there's only like 6 of them I have right now in there; though I think it breaks the player hallucination for some reason. I'll fix it though).

3. I've had another person complain about the default values being wonky. I'm assuming you mean the insanity effect almost instantly goes away when you're near people? This is actually a bug and did not realize that until this comment was made, so thank you for that.

4. This is intentional. The thought process behind the effects were to reduce the amount of information the player can make out so they're on the edge more. I will make the effects (audio and visual) toggleable / configurable in the next update. If you don't mind me asking though, does the audio effect bother you at all?

5. I would do this, but Insanity Meter already does this. No need to reinvent the wheel in my opinion.

**Comment by TaroooLvr** [comment]
4. yea it kinda bothers me and most of our friends. 

overall, we wanna use this mod because of the hallucinations, they are too good when we first played with the mod.

**Comment by Hardeh** [comment]
Dunno, for me with default config mod spams with hallucinations almost right away as you are alone, making it annoying and not scary. I'd prefer that hallucinations start appearing after some minutes while being alone in the dark, and not after 20 seconds. I even tried tweak config, divided sanity drain by 3 in all cases, but it was still way too fast.

**Comment by BudgetAirpods** [comment]
> Dunno, for me with default config mod spams with hallucinations almost right away as you are alone, making it annoying and not scary. I'd prefer that hallucinations start appearing after some minutes while being alone in the dark, and not after 20 seconds. I even tried tweak config, divided sanity drain by 3 in all cases, but it was still way too fast.

This is actually my bad. The way I coded the RNG check for the hallucinations is for them to happen every 20 seconds. An odd choice but I have increased it to 60 seconds and even made it configurable in an upcoming hotfix patch.

**Comment by BudgetAirpods** [comment]
> 4. yea it kinda bothers me and most of our friends.
> 
> overall, we wanna use this mod because of the hallucinations, they are too good when we first played with the mod.

I have added a setting allowing you to disable the panic attack audio/visual effects, as well as lowered the volume of the drone/stinger sound effects drastically and added a setting for them as well if they're too quiet for you now.

**Comment by BudgetAirpods** [comment]
> Could you explain the sanity loss scaling? I have it at 5, but I usually play with 8 people. If I set it at 8 will the players below that experience sanity loss less? We're still experiencing sanity loss at great heights even tho I configured them to lose them minimally. ie. all types of sanity is configured and gained by 0.05.
> 
> But I feel like the scaling takes more. A bit confused. Still love the mod, but everyone goes bonkers after 5 mins and even staying together.

The way the scaling works is it takes the base value of a condition (Example: Being alone would cause your sanity loss speed to be 0.26), and multiplies it by the square root of the number of players connected. To avoid ruining lobbies of 10+, I added a tweakable cap for it to stop considering other players, of which you can change if you really want to.

This means if you are in a lobby of 8 and you set the cap at 5, your new sanity loss speed should be ~0.5, and if you have it at 8, your speed should be ~ 0.7. Keep in mind this only affects _YOUR_ sanity loss as the mod is client-sided.

This is just the first iteration of it and I'm planning on coming up with better solutions for scaling the sanity loss. You can disable it by setting the players to take into account to 1; which would just use the base value you have it set to.

**Comment by Ayyobee** [comment]
> > Could you explain the sanity loss scaling? I have it at 5, but I usually play with 8 people. If I set it at 8 will the players below that experience sanity loss less? We're still experiencing sanity loss at great heights even tho I configured them to lose them minimally. ie. all types of sanity is configured and gained by 0.05.
> > But I feel like the scaling takes more. A bit confused. Still love the mod, but everyone goes bonkers after 5 mins and even staying together.
> 
> The way the scaling works is it takes the base value of a condition (Example: Being alone would cause your sanity loss speed to be 0.26), and multiplies it by the square root of the number of players connected. To avoid ruining lobbies of 10+, I added a tweakable cap for it to stop considering other players, of which you can change if you really want to.
> 
> This means if you are in a lobby of 8 and you set the cap at 5 and you are alone (meaning your base sanity speed is 0.26), your new sanity loss speed should be ~0.5, and if you have it at 8, your speed should be ~ 0.7. Keep in mind this only affects _YOUR_ sanity loss as the mod is client-sided.
> 
> This is just the first iteration of it and I'm planning on coming up with better solutions for scaling the sanity loss. You can disable it by setting the players to take into account to 1; which would just use the base value you have it set to.

I appreciate it!  Please keep improving and can't wait to see much more!  I've started modding myself but atm I'm just modeling/rigging and animating.

**Comment by BudgetAirpods** [comment]
> ###
> **[Ayyobee](/Ayyobee) ** commented [Jan 16, 2024](#issuecomment-1894628371) •

Yep that's correct! Every time you experience one it just selects a random debuff (of which I'm also working on).

I hope your modding journey is fun! This is literally my first mod and I've learned so much from it.
